### PR TITLE
Adding possibly missing indices

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Optimization

## Description

This is meant as a conversation.  Throughout the code-base we have have
`Tag.order(hotness_score: :desc)` and with PR #16322 we'll also have
`Tag.order(taggings_count: :desc)`.

My understanding is that if we have sorting, we might want to consider
an index.  I put this forward as a conversation with a possible quick win.

## Related Tickets & Documents

Related to #16322

## QA Instructions, Screenshots, Recordings

None

### UI accessibility concerns?

None

## Added/updated tests?

- [x] No, and this is why: we don't test presence of indices

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

**Review by jgaskins — APPROVED** [review]
I also noticed these were missing when evaluating one of your other PRs this week, but I didn't say anything because other filters on those queries seemed to be scoping the result sets down enough that an index might not even be engaged by the query. There is a cost to engaging an index in a query and Postgres tries to avoid paying that cost until it narrows the result set down by ~80-90% (seriously) and only when it thinks the result set would be larger than a certain threshold. It's also possible that it could change the query plan such that other indexes wouldn't be engaged.

Either way, I'm curious to see what happens. :100: Once this is merged I'll reset the index usage stats on Postgres so we can track current index usage.

**Comment by jeremyf** [comment]
> I also noticed these were missing when evaluating one of your other PRs this week, but I didn't say anything because other filters on those queries seemed to be scoping the result sets down enough that an index might not even be engaged by the query. There is a cost to engaging an index in a query and Postgres tries to avoid paying that cost until it narrows the result set down by ~80-90% (seriously) and only when it thinks the result set would be larger than a certain threshold. It's also possible that it could change the query plan such that other indexes wouldn't be engaged.

tl;dr - databases are magical!

**Review by citizen428 — APPROVED** [review]
